bitnami/redis/templates/scripts-configmap.yaml

```
@@ -54,7 +54,7 @@ data:
         {{- if .Values.useHostnames }}
         echo "${full_hostname}"
         {{- else }}
-        getent hosts "${full_hostname}" | awk '{ print $1 ; exit }'
+        until getent hosts "${full_hostname}"; do sleep 5; done | awk '{ print $1 ; exit }';
```

It could become an infinite loop if the name cannot be registered. I think a retry limit should be included and fail if the limit is reached. I mean something similar to this piece of code (not tested):
```suggestion
        let retry=0
        until [[ retry -lt 5 ]] && getent hosts "${full_hostname}"; do ((retry+=1)); sleep 5; done | grep . | awk '{ print $1 ; exit }';
```

NOTE: `grep` should fail if getent doesn't print anything.

Please keep in mind this delay in the readiness and liveness checks.

---

Yes, I thought about it, and agree that the retry limit is still needed, but let's make it a parameter because the amount of retries depends on different DNS/K8s configuration options.
For example, without adding Route53 zone to CoreDNS in a test environment, getent gives me an external IP in 5 minutes, and with CoreDNS set up in less than 1 minute